desc/objectives/_geometry.py

```diff
@@ -538,37 +526,37 @@ def build(self, eq=None, use_jit=True, verbose=1):
             warnings.warn("Plasma grid includes interior points, should be rho=1")
 
         self._dim_f = surface_grid.num_nodes
-        self._data_keys = ["R", "phi", "Z"]
-        self._args = get_params(
-            self._data_keys,
-            obj="desc.equilibrium.equilibrium.Equilibrium",
-            has_axis=plasma_grid.axis.size or surface_grid.axis.size,
-        )
+        self._equil_data_keys = ["R", "phi", "Z"]
+        self._surface_data_keys = ["x"]
 
         timer = Timer()
         if verbose > 0:
             print("Precomputing transforms")
         timer.start("Precomputing transforms")
 
-        self._surface_coords = self._surface.compute(
-            "x", grid=surface_grid, basis="xyz"
-        )["x"]
-        profiles = get_profiles(
-            self._data_keys,
+        equil_profiles = get_profiles(
+            self._equil_data_keys,
             obj=eq,
             grid=plasma_grid,
-            has_axis=plasma_grid.axis.size or surface_grid.axis.size,
+            has_axis=plasma_grid.axis.size,
         )
-        transforms = get_transforms(
-            self._data_keys,
+        equil_transforms = get_transforms(
+            self._equil_data_keys,
             obj=eq,
             grid=plasma_grid,
-            has_axis=plasma_grid.axis.size or surface_grid.axis.size,
+            has_axis=plasma_grid.axis.size,
+        )
+        surface_transforms = get_transforms(
+            self._surface_data_keys,
+            obj=self._surface,
+            grid=surface_grid,
+            has_axis=surface_grid.axis.size,
         )
         self._constants = {
-            "transforms": transforms,
-            "profiles": profiles,
-            "surface_coords": self._surface_coords,
+            "equil_transforms": equil_transforms,
+            "equil_profiles": equil_profiles,
+            "surface_transforms": surface_transforms,
+            "quad_weights": surface_grid.weights * jnp.sqrt(surface_grid.num_nodes),
         }
```

Comment:
Do we have a list somewhere in the code or docs of what all the allowed constants are?

Reply:
this is probably something specific to each objective

Reply:
yeah, each objective can define their own constants, there's no master list